common/note/note_test.go

```diff
@@ -130,151 +41,67 @@ func TestNew(t *testing.T) {
 				require.NoError(t, err)
 				require.NotNil(t, got)
 				require.Equal(t, tt.value, got.Value)
-				require.NotZero(t, got.ID)
+				require.NotNil(t, got.Preimage)
+				require.Len(t, got.Preimage, 32)
 			}
 		})
 	}
 
 	// Test for uniqueness of IDs
-	t.Run("Unique IDs", func(t *testing.T) {
-		idSet := make(map[uint64]bool)
+	t.Run("Unique preimage", func(t *testing.T) {
+		preimageSet := make(map[string]bool)
 		for i := 0; i < 1_000_000; i++ {
 			data, err := note.New(100)
 			require.NoError(t, err)
-			require.False(t, idSet[data.ID], "Generated duplicate ID: %v", data.ID)
-			idSet[data.ID] = true
+			require.False(t, preimageSet[hex.EncodeToString(data.Preimage)], "Generated duplicate preimage: %x", data.Preimage)
+			preimageSet[hex.EncodeToString(data.Preimage)] = true
 		}
 	})
 }
 
-func TestNoteRoundtrip(t *testing.T) {
-	tests := []struct {
-		name string
-		note note.Note
-	}{
-		{
-			name: "Valid note",
-			note: note.Note{
-				Data:      note.Data{ID: 12345678901234567890, Value: 100},
-				Signature: []byte("test signature"),
-			},
-		},
-		{
-			name: "Note with nil signature",
-			note: note.Note{
-				Data:      note.Data{ID: 67899, Value: 200000},
-				Signature: nil,
-			},
-		},
-		{
-			name: "Note with large signature",
-			note: note.Note{
-				Data:      note.Data{ID: 67899, Value: 200},
-				Signature: make([]byte, 1000),
-			},
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			serialized := tt.note.Serialize()
-
-			var deserialized note.Note
-			err := deserialized.Deserialize(serialized)
-			require.NoError(t, err)
-			require.Equal(t, tt.note.Data.ID, deserialized.Data.ID)
-			require.Equal(t, tt.note.Data.Value, deserialized.Data.Value)
-			require.Equal(t, tt.note.Signature, deserialized.Signature)
-		})
-	}
-}
-
 func TestNewFromString(t *testing.T) {
 	tests := []struct {
-		str           string
-		expectedID    uint64
-		expectedValue uint32
+		str              string
+		expectedPreimage []byte
```

Comment:
let's use hex strings, easier to read
```suggestion
		expectedPreimage string
```